### PR TITLE
tests: more fixtures, consolidated dates, better instructions

### PR DESCRIPTION
**Pull request by Justintime50**
# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

# Description

The goal of this PR is to make the test suite much more bullet proof without the need to manually alter fixture data in code as well as making it easier for 3rd party contributors to test this library.

- Cleaned up various tests
- Provided more fixtures to DRY up tests
- Consolidated dates in fixtures
- Removed the hard-coded child user ID
- Added helpful instructions to the README on how to test this library (from a 3rd party contributor perspective)

TODO
- Add all missing fixtures
- Strip report tests
- Fix remaining TODOs in code
- Re-record cassettes

# Testing

### CODE REVIEW

File: EasyPost.Tests/Fixture.cs [hunk 1]
```diff
@@ -393,6 +400,7 @@ public static class Fixture
             get
             {
                 return new Dictionary<string, object>
+                // TODO: Consolidate the date to a variable and reuse the same date variable in both fields below
```
Comment (Justintime50) [inline EasyPost.Tests/Fixture.cs:403]: Remove this TODO (create a single data variable and reuse it in both datetime fields below.

File: EasyPost.Tests/Fixture.cs [hunk 2]
```diff
@@ -10,24 +10,31 @@ public static class Fixture
...
-        // If ever these need to change due to re-recording cassettes, simply increment this date by 1
-        public const string ReportEndDate = "2022-02-03";
+        // If you need to re-record cassettes, increment this date by 1
+        public const string ReportDate = "2022-04-12";
 
-        public static string RandomUrl => $"https://{Guid.NewGuid().ToString().Substring(0, 8)}.com";
+        public static string WebhookUrl => "http://example.com";
+
+        // This is the USPS carrier account ID that comes with your EasyPost account by default and should be used for all tests
+        public static string UspsCarrierAccountId
```
Comment (Justintime50) [inline EasyPost.Tests/Fixture.cs:29]: Nit: Can we keep the original order of this in the file (smaller diff, faster eye scanning across libs)

File: EasyPost.Tests/cassettes/user/retrieve.json [hunk 3]
```diff
@@ -1,48 +0,0 @@
-[
```
Comment (Justintime50) [inline EasyPost.Tests/cassettes/user/retrieve.json:1]: This cassette needs to be re-recorded.

**Review by nwithan8 — APPROVED** [review]
LGTM, me.

**Review by jchen293 — APPROVED** [review]

**Pull request merged**